examples/switch-state.rs

```diff
@@ -0,0 +1,19 @@
+//! Let's you easily try out the tokenizer with e.g.
+//! printf '<style><b>Hello world!</b></style>' | cargo run --example=switch-state
+use html5gum::{BufReadReader, Token, Tokenizer};
+use std::io::stdin;
+
+fn main() {
+    let stdin = stdin();
+    let mut tokenizer = Tokenizer::new(BufReadReader::new(stdin.lock()));
+
+    while let Some(token) = tokenizer.next() {
+        let token = token.unwrap();
+        println!("{:?}", token);
+
+        if let Token::StartTag(start_tag) = token {
```

Comment:
is this actually correct usage of the helper function? I'm reading the spec and i can't find a section that actually says you're supposed to override the tokenizer state for every start tag?

I'd add some more information in doc comments how this functionally differs from the other example you added, + whether this is spec-compliant